src/backend/database_models/agent.py

```diff
@@ -65,10 +80,21 @@ class Agent(Base):
         back_populates="agents",
         overlaps="deployments,models,agents,agent,agent_deployment_associations,model",
     )
+    associated_tools = relationship(
+        "Tool",
+        secondary="agent_tool",
+        back_populates="agents",
+        overlaps="tools,agents,agent,agent_tool_associations,tool",
+    )
+
     agent_deployment_associations = relationship(
         "AgentDeploymentModelAssociation", back_populates="agent"
     )
 
+    agent_tool_associations = relationship(
+        "AgentToolAssociation", back_populates="agent"
+    )
+
```

Comment:
What is the reason to have the `associated_tools` and `agent_tool_associations`? Maybe I'm misunderstanding something but I think they are basically the same thing (?)

Reply:
These are different entities, associated_tools is a list of tools, the agent_tool_associations is a list of associations